pymc3/distributions/distribution.py

```diff
@@ -167,7 +167,14 @@ def getattr_value(self, val):
         if isinstance(val, tt.TensorVariable):
             return val.tag.test_value
 
-        if isinstance(val, tt.sharedvar.TensorSharedVariable):
+        if isinstance(
+            val,
+            (
+                tt.sharedvar.TensorSharedVariable,  # pm.Data or theano.shared tensor
```

Comment:
Just realized that instead of having the 3 specific types we can simply check `isinstance(val, tt.sharedvar.SharedVariable)`